tools/data/textrecog/ilst_converter.py

```diff
@@ -0,0 +1,265 @@
+
+    Args:
+        gt_file (str): The path to ground-truth
+        img_info (dict): The dict of the img and annotation information
+
+    Returns:
+        img_info (dict): The dict of the img and annotation information
+    """
+    obj = ET.parse(gt_file)
+    root = obj.getroot()
+    anno_info = []
+    for object in root.iter('object'):
+        word = object.find('name').text
+        x1 = int(object.find('bndbox').find('xmin').text)
+        y1 = int(object.find('bndbox').find('ymin').text)
+        x2 = int(object.find('bndbox').find('xmax').text)
+        y2 = int(object.find('bndbox').find('ymax').text)
+
+        x = max(0, min(x1, x2))
+        y = max(0, min(y1, y2))
+        w, h = abs(x2 - x1), abs(y2 - y1)
+        bbox = [x, y, x + w, y, x + w, y + h, x, y + h]
+        anno = dict(bbox=bbox, word=word)
+        anno_info.append(anno)
+
+    img_info.update(anno_info=anno_info)
+
+    return img_info
+
+
+def split_train_val_list(full_list, val_ratio):
+    """Split list by val_ratio
+
+    Args:
+        full_list (list): List to be splited
+        val_ratio (float): Split ratio for val set
+
+    return:
+        list(list, list): Train_list and val_list
+    """
+
+    n_total = len(full_list)
+    offset = int(n_total * val_ratio)
+    if n_total == 0 or offset < 1:
+        return [], full_list
+    val_list = full_list[:offset]
+    train_list = full_list[offset:]
+    return [train_list, val_list]
+
+
+def generate_ann(root_path, image_infos, preserve_vertical, val_ratio, format):
+    """Generate cropped annotations and label txt file.
+
+    Args:
+        root_path (str): The root path of the dataset
+        split (str): The split of dataset. Namely: training or test
+        image_infos (list[dict]): A list of dicts of the img and
+            annotation information
+        preserve_vertical (bool): Whether to preserve vertical texts
+        val_ratio (float): Split ratio for val set
+        format (str): Using jsonl(dict) or str to format annotations
+    """
+
+    assert val_ratio <= 1.
+
+    if val_ratio:
+        image_infos = split_train_val_list(image_infos, val_ratio)
+        splits = ['training', 'val']
+
+    else:
+        image_infos = [image_infos]
+        splits = ['training']
+
+    for i, split in enumerate(splits):
+        dst_image_root = osp.join(root_path, 'crops', split)
+        ignore_image_root = osp.join(root_path, 'ignores', split)
+        dst_label_file = osp.join(root_path, f'{split}_label.{format}')
+        os.makedirs(dst_image_root, exist_ok=True)
+
+        lines = []
+        for image_info in image_infos[i]:
+            index = 1
+            src_img_path = osp.join(root_path, 'imgs', image_info['file_name'])
+            image = mmcv.imread(src_img_path)
+            src_img_root = image_info['file_name'].split('.')[0]
+
+            for anno in image_info['anno_info']:
+                word = anno['word']
+                dst_img = crop_img(image, anno['bbox'])
```

Comment:
Please specify the padding params to 0, the default padding params introduce too much noise for this dataset.
`dst_img = crop_img(image, anno['bbox'], 0, 0)`

<img width="232" alt="1648454933(1)" src="https://user-images.githubusercontent.com/45810070/160354504-f78b4332-2dd8-4af5-8b46-a3f56093d1a3.png">